### PR TITLE
Add ability to give EventBridge events custom names

### PR DESCRIPTION
**Pull request by ROSeaboyer**
Closes: #11360

### CODE REVIEW

File: test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js [hunk 1]
```diff
@@ -564,6 +590,11 @@ describe('EventBridgeEvents', () => {
         expect(deadLetterConfigRuleTarget.DeadLetterConfig).to.have.property('Arn');
       });
 
+      it('should correctly set event name when set', () => {
+        const customNameTarget = getRuleResourceEndingWith(cfResources, 'test').Properties;
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js:594]: All new tests should be configured purely with `runServerlss` util. See bottom of this test file, I believe you can add there custom name case.

Also check: https://github.com/serverless/serverless/tree/main/test#unit-tests

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @ROSeaboyer. All looks good, just tests need to be reconfigured.

File: test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js (hunk 1 above)
Comment (ROSeaboyer) [reply test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js]: @medikoo tests updated; let me know if there's still any issues

File: test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js [hunk 2]
```diff
@@ -688,3 +688,42 @@ describe('EventBridgeEvents', () => {
...
+
+    before(async () => {
+      const { awsNaming, cfTemplate } = await runServerless({
+        fixture: 'function',
+        command: 'package',
+        configExt: {
+          functions: {
+            customName: {
+              handler: 'basic.handler',
+              name: 'custom-name-lambda',
```
Comment (medikoo) [inline test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js:704]: `runServerless` calls are expensive, so maybe let's add this test case to run that's configured at L590

Comment (ROSeaboyer) [reply test/unit/lib/plugins/aws/package/compile/events/event-bridge/index.test.js]: Whoops, fixed!

**Review by medikoo — APPROVED** [review]
Thank you @ROSeaboyer !

**Pull request merged**